### PR TITLE
Add promote posts button to stats entries, with feature flag

### PR DESCRIPTION
**Pull request by j6ll**
#### Proposed Changes

Hi Team, this is my first calypso PR and part of a series to add entrypoints in calypso for a new post promotion widget. Which will allow wordpress.com users to pay for advertising and serve the posts as ads on various A8C products.

Info specifically on these tasks here (pbpazC-3cN-p2) and the wider project thread here (pbpazC-2Pd-p2)

The majority of this feature is done outside of WP.com as a standalone widget, here we only need to add buttons to Calypso to show the widget and to pass it the site & post id's.

Note: We will be iterating on a bunch of PRs to complete this feature, but to keep these smaller we will be merging one-by-one under a feature flag.

Here are our provided designs:

Screenshot 2022-07-28 at 11 05 39

And here is the screenshot in my development environment:

#### Testing Instructions

-  Enable the feature flag `promote-posts` either on [startup](https://wpcalypso.wordpress.com/devdocs/config/README.md#feature-flags:~:text=Testing%20Feature%20Flags,reader%20yarn%20start) or [via url](https://wpcalypso.wordpress.com/devdocs/config/README.md#feature-flags:~:text=Testing%20Feature%20Flags%20via%20URLs,and%20disables%20feature%20bar.) using the `?flags=promote-post` query string on the following urls
-  Visit the stats page for your favourite blog for example: `http://calypso.localhost:3000/stats/day/something.wordpress.com`
-  In the posts and pages section, hover over a post and you should see a "speaker" icon like in the designs abover
Screenshot 2022-07-28 at 11 20 45
-  clicking the button should open the widget.  However, this is not in production yet so without setting up your environment you can check the network tab and notice that it's trying to hit the script `widget.js` - this will repeatedly fail.      In a working dev environment - it only ever loads the script once.
-  Test the above Steps without the feature flag enabled,  ensure `widget.js` is not loaded in the f12 -> network tab and that the speaker button does not appear on hover.

**Without local instance of the widget running**
Screenshot 2022-07-28 at 11 29 46

**With local instance of the widget running**
Screenshot 2022-07-28 at 11 33 47

#### Pre-merge Checklist

- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have you sent any new strings for translation (PCYsg-1vr-p2) ASAP?

Related to #

### CODE REVIEW

**Review by benzhu56 — COMMENTED** [review]
Works great and code LG.
I happened to notice one thing, that a `page` would also show up in the list, however, it would have an id of 0. Not sure if it has an actual id and are we able to get the actual id.
E.g in following example, the top page has an id of 0, (I printed it out in console).
image

**Comment by j6ll** [comment]
>  I happened to notice one thing, that a `page` would also show up in the list, however, it would have an id of 0. 

Thanks @benzhu56 spotted this too.  My thoughts were that we should still send show the user the promote button - but send them to a blank widget (not pre-populated).   Then we need the widget just to handle `0` or invalid post Id's properly - not show it as a validation error on the creative screen..

I created a follow on ticket for that here: (350-gh-tumblr/wordads-picard).    The other part of this longer term would be the post picker in the widget - I think this is maybe being discussed but not ticketed yet.

File: client/state/stats/lists/test/utils.js [hunk 1]
```diff
@@ -572,6 +572,7 @@ describe( 'utils', () => {
 					)
 				).toEqual( [
 					{
+						id: 0,
```
Comment (gravityrail) [inline client/state/stats/lists/test/utils.js:575]: Just a heads up that the home page post ID isn't always going to be 0 (you can switch any post with a post_type of `page` to be the home page too, and these will all have an ID)

Also there are other URLs, like taxonomy pages or the blog page, that also might not have IDs.

**Review by gravityrail — APPROVED** [review]
This LGTM

**Comment by gravityrail** [comment]
I have tested this and, while it's clearly part of a work-in-progress, it doesn't break anything and is required to unblock further work so I'm going to merge it.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7462764

Thank you @jamesgill3 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7462764).

**Pull request merged**